src/battle_game/core/units/Random_npc.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include "battle_game/core/unit.h"
+#include "battle_game/core/units/tiny_tank.h"
+
+namespace battle_game::unit {
+class RandomNPC : public Tank {
+ public:
+  RandomNPC(GameCore *game_core, uint32_t id, uint32_t player_id);
+  void Render() override;
+  void Update() override;
+  [[nodiscard]] bool IsHit(glm::vec2 position) const override;
+  [[nodiscard]] float GetDamageScale() const override;
+  [[nodiscard]] float BasicMaxHealth() const override;
+
+ protected:
+  void Fire();
+  void RandomMove();
+  void RandomTurretRotate();
+  [[nodiscard]] const char *UnitName() const override;
+  [[nodiscard]] const char *Author() const override;
+  int current_steps;
+  int current_direction;
```

Comment:
Had better rename the protected/private variables like: `current_steps_`

Reply:
modified in my newest commmit([Merge branch 'Yao-class-cpp-studio:main' into main](https://github.com/Yao-class-cpp-studio/battle_game/pull/329/commits/e005e172e11da6b009f3d796a39bef8237989fe2)